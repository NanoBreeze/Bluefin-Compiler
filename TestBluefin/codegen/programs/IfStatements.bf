/*
Verifies that if statements work correctly. Include if/else, if, and recursive ifs
*/

void foo(int a) {

	if (a > 5) {
		a + 5;
		if (a > 6) { a + 6; }
		else { a - 6; }
		a + 7;
	}
	else {
		if (a < 8) {
			a - 8;	
			if (a < 9) {}
		}
		a - 5;
	}

	a + 10;

	
	if (a > 50) {}

	a + 60;
	
}