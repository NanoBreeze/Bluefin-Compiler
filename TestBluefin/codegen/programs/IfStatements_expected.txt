; ModuleID = 'IfStatements.bf'
source_filename = "IfStatements.bf"

define void @foo(i32 %a) {
entry:
  %cmpSGTtmp = icmp sgt i32 %a, 5
  br i1 %cmpSGTtmp, label %then, label %else

then:                                             ; preds = %entry
  %addtmp = add i32 %a, 5
  %cmpSGTtmp4 = icmp sgt i32 %a, 6
  br i1 %cmpSGTtmp4, label %then1, label %else2

else:                                             ; preds = %entry
  %cmpSLTtmp = icmp slt i32 %a, 8
  br i1 %cmpSLTtmp, label %then7, label %merge8

merge:                                            ; preds = %merge8, %merge3
  %addtmp14 = add i32 %a, 10
  %cmpSGTtmp17 = icmp sgt i32 %a, 50
  br i1 %cmpSGTtmp17, label %then15, label %merge16

then1:                                            ; preds = %then
  %addtmp5 = add i32 %a, 6
  br label %merge3

else2:                                            ; preds = %then
  %subtmp = sub i32 %a, 6
  br label %merge3

merge3:                                           ; preds = %else2, %then1
  %addtmp6 = add i32 %a, 7
  br label %merge

then7:                                            ; preds = %else
  %subtmp9 = sub i32 %a, 8
  %cmpSLTtmp12 = icmp slt i32 %a, 9
  br i1 %cmpSLTtmp12, label %then10, label %merge11

merge8:                                           ; preds = %merge11, %else
  %subtmp13 = sub i32 %a, 5
  br label %merge

then10:                                           ; preds = %then7
  br label %merge11

merge11:                                          ; preds = %then10, %then7
  br label %merge8

then15:                                           ; preds = %merge
  br label %merge16

merge16:                                          ; preds = %then15, %merge
  %addtmp18 = add i32 %a, 60
  ret void
}
